src/core/core.layouts.js

```diff
@@ -48,10 +98,137 @@ function findMaxPadding(boxes) {
 	};
 }
 
-function addSizeByPosition(boxes, size) {
-	helpers.each(boxes, function(box) {
-		size[box.position] += box.isHorizontal() ? box.height : box.width;
-	});
+function addSizeByPosition(size, boxes) {
+	var result = extend({}, size);
+	var i, ilen, layout, box;
+	for (i = 0, ilen = boxes.length; i < ilen; ++i) {
+		layout = boxes[i];
+		box = layout.box;
+		result[box.position] += layout.isHorizontal ? box.height : box.width;
+	}
+	return result;
+}
+
+function calculateMinSizes(boxes, params) {
+	var chartArea = params.chartArea;
+	var i, ilen, layout, box, minSize;
+
+	for (i = 0, ilen = boxes.length; i < ilen; ++i) {
+		layout = boxes[i];
+		box = layout.box;
+
+		layout.isHorizontal = layout.box.isHorizontal();
+		if (layout.isHorizontal) {
+			minSize = box.update(box.fullWidth ? params.width : chartArea.w, params.hBoxMaxHeight);
+			chartArea.h -= minSize.height;
+		} else {
+			minSize = box.update(params.vBoxMaxWidth, chartArea.h);
+			chartArea.w -= minSize.width;
+		}
+		layout.minSize = minSize;
+	}
+	return addSizeByPosition(params.padding, boxes);
+}
+
+function fitHorizontalBoxes(boxes, params, size) {
+	var maxPadding = params.maxPadding;
+	var i, ilen, layout, box, margin;
+
+	for (i = 0, ilen = boxes.length; i < ilen; ++i) {
+		layout = boxes[i];
+		box = layout.box;
+
+		margin = {
+			left: Math.max(size.left, maxPadding.left),
+			right: Math.max(size.right, maxPadding.right),
```

Comment:
I was going to comment that this seemed wrong because `size.right` was the right edge of the box until I realized that it's just the total width of all boxes in the `'right'` position. Perhaps it's worthy of a comment? Or maybe renaming `size` to be a bit clearer

Reply:
I suggested renaming as well: https://github.com/chartjs/Chart.js/pull/6304#discussion_r288301482

Reply:
renamed to `outerDims`